src/nvim/fileio.c

```diff
@@ -6147,6 +6147,11 @@ void ex_doautoall(exarg_T *eap)
     /* restore the current window */
     aucmd_restbuf(&aco);
 
+    if (aco.new_curbuf->terminal) {
+      terminal_resize(aco.new_curbuf->terminal, aco.save_curwin->w_width,
+                      aco.save_curwin->w_height);
+    }
```

Comment:
Yes, or maybe even more general.. because the actual problem is a terminal buffer shown multiple times in different windows.

E.g. http://imgur.com/MTJIB2p

Here, the terminal buffer was created in the bottom right window. Then I switched to that buffer in the left window and can only scroll to the middle, since it's the height of the window the terminal buffer was created in.

So.. it should probably done in `win_enter()` instead?